core_contracts/governance/governance.py

```diff
@@ -134,19 +150,35 @@ def castVote(self, name: str, vote: bool) -> None:
         snapshot = proposal.vote_snapshot.get()
         baln = self.create_interface_score(self.addresses['baln'], BalancedInterface)
         stake = baln.stakedBalanceOfAt(sender, snapshot)
+        if stake == 0:
+            revert(f'Balanced tokens needs to be staked to cast the vote.')
```

Comment:
Will need to base this on the total voting power including LP'd BALN once merged with the branch that includes LP'd BALN.